app/packs/src/utilities/MathUtils.js

```diff
@@ -56,9 +56,42 @@ const formatBytes = (bytes, decimals = 2) => {
   return `${parseFloat((bytes / (k ** i)).toFixed(dm))} ${sizes[i]}`;
 };
 
+function parseNumericString(numberString) {
+  if (typeof numberString !== 'string') {
+    return NaN;
+  }
+
+  const numberIsNegative = numberString.startsWith('-');
+  let sanitizedNumberString = numberString;
+
+  // Remove all characters that aren't digits, commas, or periods.
+  sanitizedNumberString = sanitizedNumberString.replace(/[^0-9,.]/g, '');
+  if (sanitizedNumberString === '') {
+    return NaN;
+  }
+
+  // Decimal separator can be comma or period. Convert to period.
+  sanitizedNumberString = sanitizedNumberString.replaceAll(',', '.');
+  // Keep only final (non-terminal) period under the assumption that it's meant as the decimal separator.
+  // Assume that preceding periods were meant as thousands separators.
+  const finalPeriodIndex = sanitizedNumberString.lastIndexOf('.');
+  if (finalPeriodIndex !== -1) {
+    sanitizedNumberString = `${sanitizedNumberString.slice(0, finalPeriodIndex).replaceAll('.', '')
+    }.${
+      sanitizedNumberString.slice(finalPeriodIndex + 1)}`;
+  }
+
+  if (numberIsNegative) {
+    sanitizedNumberString = `-${sanitizedNumberString}`;
+  }
```

Comment:
I would split this block into three sub-functions:
Advantage: you can perhaps get rid of the comments and it is clear in the later refactoring that essentially three steps are applied: 
1. remove all non-numeric characters
2. replace commas with periods and declare only the last period as a decimal point.
3. set the sign

Its a matter of taste, you might argue that you want to have this behavior as close as possible together. I am also fine with that in this case.

Reply:
I see your point. In other cases, I'd probably agree, however, in this case I'd prefer not to fragment the function because:

- I don't see much utility / re-usability in the sub-functions. At least at the moment. I'd defer extraction until I (or someone else) wants to re-use the logic partially.
- It's a matter of taste, but to my eyes, fragmenting an already short(ish) function hurts readability rather than helps it.
Even if the sub-functions are well-named, I'd still read them (which requires jumping around) just to make sure they're doing what I think they do.
- I'm not interested in testing any of the sub-functions. Atm I only care for their aggregate behavior. I.e., I'm fine treating the function as a black box. In fact, I *want* it to be a black box (hiding complexity).

Reply:
As i said in my first comment, i agree, it's a matter of taste. From my point of view, the readability and understandability would increase, because I can read through the algorithm of this function, without having to care about the details of the implementation. It would be only "private" helper functions, so it would be still fine to just test the main function as a blackbox. 

 In this specific case, the algorithm is quite clear, so I don't have problems with keeping it as it is.